_posts/2024-02-01-wildfly-glow.adoc

```diff
@@ -0,0 +1,504 @@
+
+==== Handling of 2 execution contexts
+
+* bare-metal (the default).
+* cloud, to execute on Openshift and/or Kubernetes. In addition to fine tune the server configuration for cloud execution, 
+it discovers and advertises the WildFly env variables usable to configure the server at startup.
+
+==== Handling of datasources
+
+* Identify missing datasources and suggest datasources that you can use to connect to databases.
+
+==== Knowledge of extra Feature-packs
+
+A centralized knowledge (located in the https://github.com/wildfly/wildfly-galleon-feature-packs[wildfly-galleon-feature-packs] github project) 
+of extra Galleon feature-packs compatible with WildFly. The WildFly features defined in these extra feature-packs 
+are included in the provisioned server when WildFly Glow detects a need for them. 
+
+The set of extra features supported are:
+
+* https://github.com/wildfly-extras/wildfly-cloud-galleon-pack[Cloud features]
+* Support for the major https://github.com/wildfly-extras/wildfly-datasources-galleon-pack[datasources] (PostgreSQL, MySQL, MariaDB, MSSQLServer, Oracle)
+* https://github.com/keycloak/keycloak[Keycloak] SAML
+* https://github.com/wildfly-extras/wildfly-grpc-feature-pack[GRPc]
+* https://github.com/wildfly-extras/wildfly-myfaces-feature-pack[MyFaces]
+* https://github.com/wildfly-extras/wildfly-graphql-feature-pack[Microprofile-GraphQL]
+
+==== Support for WildFly Preview
+
+WildFly Glow allows you to choose to provision a WildFly Preview server instead of a WildFly server.
+ 
+=== How WildFly Glow operates
+
+WildFly Glow is used from provisioning tooling: WildFly Glow command line tool or WildFly Maven Plugin (starting with version 5.0.0 Beta).
+
+Java classes and file descriptors located in your deployments are scanned to identify the required set of galleon layers. 
+
+* It leverages Galleon provisioning artifacts (Feature-packs and Layers).
+* It relies on rules included in each Galleon Layer.
+** Rules capture the content expected inside the deployment for the layer to be required.
+** Rules express the High Availability capability of a layer.
+** Rules classify some layers as `add-on` that can be explicitly included. add-ons are advertised according to the set of layers discovered in the deployments.
+e.g.: SSL, embedded/remote JMS brokers, datasources, WildFly CLI (jboss-cli, add-users, elytron tooling, .,..).
+* It knows about High Availability profile, and will automatically include HA Galleon layers.
+
+==== WildFly Glow rules
+
+You can find documentation on the rules contained by WildFly Galleon layers in 
+this http://docs.wildfly.org/wildfly-galleon-feature-packs/[documentation].
+
+==== WildFly Glow Provisioning diagram
+
+image::wildfly-glow2.jpg[WildFly Glow provisioning diagram]
+
+== Provisioning tooling usage comparison
+
+=== WildFly Maven plugin
+
+==== Simple server
+
+===== Configuration without WildFly Glow support
+
+[source,xml]
+----
+...
+
+<feature-packs>
```

Comment:
Although we might not want to use the full plugin configuration in EVERY example (it is verbose and will probably detract from the core concept), I feel some context is missing.

Maybe in the "WildFly Maven plugin" introduction you could do something like:

The WildFly Maven Plugin is used for both the Galleon and WildFly Glow provisioning use-cases. The XML will typically look something like. We will give detailed examples of the 'Galleon or WildFly Glow specific configuration' placeholder further down.

```
<plugin>
    <groupId>org.wildfly.plugins</groupId>
    <artifactId>wildfly-maven-plugin</artifactId>
    <configuration>
        <!-- Galleon or WildFly Glow specific configuration -->
        <!-- deploys the quickstart on root web context -->
        <name>ROOT.war</name>
    </configuration>
    <executions>
        <execution>
            <goals>
                <goal>package</goal>
            </goals>
        </execution>
    </executions>
</plugin>
```